lib/control-connection.js

```diff
@@ -685,6 +694,14 @@ ControlConnection.prototype.setPeersInfo = function (initializing, err, result,
       });
       self.hosts.removeMultiple(toRemove);
     }
+    if (initializing && self.options.localDataCenter) {
+      const localDc = self.options.localDataCenter;
+
+      if (!foundDataCenters.has(localDc)) {
+        return callback(new errors.ArgumentError('localDataCenter was configured as \'' + localDc + '\', but only found' +
```

Comment:
Wasn't sure if this is the right place for this, or if we should error out or warn.  The way this currently behaves is if no hosts are found using the specified local DC, the code errors out with:

```
NoHostAvailableError: All host(s) tried for query failed. First host tried, 127.0.0.101:9042: ArgumentError: localDataCenter was configured as 'dc2', but only found hosts in data centers: [dc1]. See innerErrors.
      at whilstEnded (lib/control-connection.js:231:25)
      at next (lib/utils.js:825:14)
      at Object.whilst (lib/utils.js:819:3)
      at ControlConnection.borrowAConnection (lib/control-connection.js:205:9)
      at Array.getConnectionTask (lib/control-connection.js:383:12)
      at /home/tolbertam/Documents/Projects/nodejs-driver/lib/utils.js:692:21
      at process._tickCallback (internal/process/next_tick.js:61:11)
```

Reply:
Good call, validating it here prevents duplicating the logic on the LBPs.